README.md

```diff
@@ -15,23 +44,28 @@ system to make contributing to this repository easier.
 
 ### Exercises
 
-All exercises have the following structure:
+All exercises are registered in the file
+- `config.json`: Every exercise has a unique name and a difficulty. The sequence order is also the order in which the exercises are to be solved.
+
+Each exercise has the following structure:
 
 - `stack.yaml` has just one line specifying the current
 [Stack snapshot](https://www.stackage.org/snapshots). We use the same
 resolver for all the exercises.
 - `package.yaml` is a file in the [hpack](https://github.com/sol/hpack#readme)
 format that has all dependencies and build instructions for an exercise.
-- `src/ModuleName.hs` is a *stub solution*.
-- `src/examples/success-foo/package.yaml` has dependencies for the example solution named `foo`.
-- `src/examples/success-foo/src/ModuleName.hs` is the source code of the sample solution.
-- `test/Tests.hs` is the test suite.
-- `HINTS.md` is an optional file containing instructions and/or hints.
+- `src/ModuleName.hs` is a *stub solution* that compiles together with the test suite. It should be as general as possible in order to not exclude any possible solutions. It should take Haskell specifics into account (for example use `Maybe` instead of a dummy return value). It should not contain any comments (people might forget to remove them), you can use the hints file instead.
```

Comment:
I would separate the description about the exercise's structure from the recomendations about how to design a good test suite.